src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,108 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final float LOAD_FACTORY = 0.75f;
+    private int initialCapacity = 16;
+    private int threshold;
+    private Node<K, V>[] table;
+    private int size;
+
+    @SuppressWarnings("unchecked")
+    public MyHashMap() {
+        this.table = (Node<K, V>[]) new Node[initialCapacity];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        threshold = (int) (initialCapacity * LOAD_FACTORY);
+        if (size >= threshold) {
+            resize();
+        }
+        int index = Math.abs(hash(key)) % table.length;
+        var current = table[index];
+        if (table[index] == null) {
+            table[index] = new Node<>(index, key, value, null);
+            size++;
+        } else {
+            while (current != null) {
+                if (key == null && current.key == null) {
+                    current.value = value;
+                    return;
+                }
+                if (key != null && key.equals(current.key)) {
+                    current.value = value;
+                    return;
+                }
+                if (current.next == null) {
+                    current.next = new Node<>(index, key, value, null);
+                    size++;
+                    return;
+                }
+                current = current.next;
+            }
+        }
     }
 
     @Override
     public V getValue(K key) {
+        int index = Math.abs(hash(key)) % table.length;
+        Node<K, V> current = table[index];
+        while (current != null) {
+            if (key == null && current.key == null) {
+                return current.value;
+            }
+            if (key != null && key.equals(current.key)) {
+                return current.value;
+            }
+            current = current.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    Node<K, V>[] resize() {
```

Comment:
Using public access modifiers everywhere. If the method has only a utility purpose and is used only inside the same class, it should not be public. Keep your code as close as possible to follow the encapsulation principle.